File: app/src/main/java/com/example/usphuong/flicks/entity/Videos.java
package com.example.usphuong.flicks.entity;

import java.util.List;

import com.example.usphuong.flicks.model.Trailer;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Videos {

    @SerializedName("id")
    @Expose
    private Integer id;
    @SerializedName("results")
    @Expose
    private List<VideoListDTO> results = null;

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public List<VideoListDTO> getResults() {
        return results;
    }

    public void setResults(List<VideoListDTO> results) {
        this.results = results;
    }

    /**
     * Small Video Data Transfer Object
     * Used to store a video list retrieved from a list of movies from TMDb
     */
    public static class VideoListDTO {

        private String id;
        private String iso6391;
        private String iso31661;
        private String key;
        private String name;
        private String site;
        private Integer size;
        private String type;

        public String getId() {
            return id;
        }

        public void setId(String id) {
            this.id = id;
        }

        public String getIso6391() {
            return iso6391;
        }

        public void setIso6391(String iso6391) {
            this.iso6391 = iso6391;
        }

        public String getIso31661() {
            return iso31661;
        }

        public void setIso31661(String iso31661) {
            this.iso31661 = iso31661;
        }

        public String getKey() {
            return key;
        }

        public void setKey(String key) {
            this.key = key;
        }

        public String getName() {
            return name;
        }

        public void setName(String name) {
            this.name = name;
        }

        public String getSite() {
            return site;
        }

        public void setSite(String site) {
            this.site = site;
        }

        public Integer getSize() {
            return size;
        }

        public void setSize(Integer size) {
            this.size = size;
        }

        public String getType() {
            return type;
        }

        public void setType(String type) {
            this.type = type;
        }
    }
}

File: app/src/main/java/com/example/flicks/entity/Videos.kt
package com.example.flicks.entity

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class Videos {
    @SerializedName("id")
    @Expose
    var id: Int? = null

    @SerializedName("results")
    @Expose
    var results: List<VideoListDTO>? = null

    /**
     * Small Video Data Transfer Object
     * Used to store a video list retrieved from a list of movies from TMDb
     */
    class VideoListDTO {
        var id: String? = null
        var iso6391: String? = null
        var iso31661: String? = null
        var key: String? = null
        var name: String? = null
        var site: String? = null
        var size: Int? = null
        var type: String? = null

    }
}